packages/features/bookings/Booker/Booker.tsx

```diff
@@ -41,6 +41,17 @@ const BookerComponent = ({
   isTeamEvent,
   entity,
 }: BookerProps) => {
+  /**
+   * Prioritize dateSchedule load
+   * Component will render but use data already fetched from here, and no duplicate requests will be made
+   * */
+  useScheduleForEvent({
+    prefetchNextMonth: false,
+    username,
+    eventSlug,
+    month,
+    duration: undefined,
+  });
```

Comment:
Calling hook right away in the render.

Reply:
It isn't intuitive as we are not using the returned value in this component. Instead, I like this approach of letting the Component render it's child and let the child component fetch `useSchedule` as it's the one who needs it
https://github.com/calcom/cal.com/pull/10588/files#diff-d8cdd443610a4483db45dcc4da6a02bcff302ad6c591aea6fa1ee8646a52aedbL168-L173

Reply:
Neither way is particularly great tbh

Reply:
Yeah I agree.

The problem actually is that `useScheduleForEvent` needs to have `eventTypeId`(and other props) set in zustand booker store. So, we are solving the problem by calling `useScheduleForEvent` here itself and providing it the values directly. If we can solve it in zustand store(bookerStore) directly that would be the best IMHO(but that wasn't possible I tried), but I feel next best thing is that the custom hook `useEvent` gives you the correct data wherever you use it. So, I fixed it in `useEvent` itself. So, once it's fixed, every component and hook would get the correct data.


Another approach might be to pass these props directly to the components needing them instead of them using `useEvent` but that might be a lot of prop drilling which feels unnecessary and counter intuitive to me.

Reply:
My guess is this will be solved once we reach react server components. But for now since this is a core component we can use this for now and let the rest keep working as usual.